sentinel-adapter/sentinel-api-gateway-adapter-common/src/main/java/com/alibaba/csp/sentinel/adapter/gateway/common/command/UpdateGatewayRuleCommandHandler.java

```diff
@@ -54,8 +57,41 @@
         String result = SUCCESS_MSG;
         List<GatewayFlowRule> flowRules = JSONArray.parseArray(data, GatewayFlowRule.class);
         GatewayRuleManager.loadRules(new HashSet<>(flowRules));
+        if (!writeToDataSource(gatewayFlowWds, new HashSet<>(flowRules))) {
```

Comment:
Here creating HashSet twice, how about optimizing it?

Reply:
ok，i will optimize it.

Reply:
@cdfive I had optimized it.